pypose/optim/kernel.py

```diff
@@ -123,3 +125,150 @@ def forward(self, input: Tensor) -> Tensor:
+
+
+
+class SoftLOne(nn.Module):
+    r"""The robust SoftLOne kernel cost function.
+
+    .. math::
+        \bm{y}_i=2\left ( \delta \sqrt{\frac{1}{{\delta{}}^{2}}+\bm{x}_i}- {{\delta{}}^{2}}\right )
+
+    where :math:`\delta` (delta) is a hyperparameter, :math:`\bm{x}` and :math:`\bm{y}` are the
+    input and output tensors, respectively.
+
+    Args:
+        delta (float, optional): Specify the SoftLOne cost. The value must be positive. Default: 1.0
+
+
+    Note:
+        The input has to be a non-negative tensor and the output tensor has the same shape with the
+        input.
+
+    Example:
+        >>> import pypose.optim.kernel as ppok
+        >>> kernel = ppok.SoftLOne()
+        >>> input = torch.tensor([0, 0.5, 1, 2, 3])
+        >>> kernel(input)
+        tensor([0.0000, 0.4495, 0.8284, 1.4641, 2.0000])
+    """
+    def __init__(self, delta: float = 1.0) -> None:
+        super().__init__()
+        assert delta > 0, ValueError("delta has to be positive: {}".format(delta))
+        self.delta1 = delta
+        self.delta2 = delta**2
+
+    def forward(self, input: Tensor) -> Tensor:
+        '''
+        Args:
+            input (torch.Tensor): the input tensor (non-negative).
+        '''
+        assert torch.all(input >= 0), 'input has to be non-negative'
+        return 2*(self.delta1*(1/self.delta2+input).sqrt()-self.delta2)
+
+class Arctan(nn.Module):
+    r"""The robust Arctan kernel cost function.
+
+    .. math::
+        \bm{y}_i=\delta ^{2}\arctan \left ( \frac{\bm{x}_i}{\delta ^{2}}\right )
+
+    where :math:`\delta` (delta) is a hyperparameter, :math:`\bm{x}` and :math:`\bm{y}` are the
+    input and output tensors, respectively.
+
+    Args:
+        delta (float, optional): Specify the Arctan cost. The value must be positive. Default: 1.0
+
+    Note:
+        The input has to be a non-negative tensor and the output tensor has the same shape with the
+        input.
+
+    Example:
+        >>> import pypose.optim.kernel as ppok
+        >>> kernel = ppok.Arctan()
+        >>> input = torch.tensor([0, 0.5, 1, 2, 3])
+        >>> kernel(input)
+        tensor([0.0000, 0.4636, 0.7854, 1.1071, 1.2490])
+    """
+    def __init__(self, delta: float = 1.0) -> None:
+        super().__init__()
+        assert delta > 0, ValueError("delta has to be positive: {}".format(delta))
+        self.delta1 = delta
+        self.delta2 = delta**2
+
+    def forward(self, input: Tensor) -> Tensor:
+        '''
+        Args:
+            input (torch.Tensor): the input tensor (non-negative).
+        '''
+        assert torch.all(input >= 0), 'input has to be non-negative'
+        return self.delta2*(input/self.delta2).arctan()
+
+
+class Tolerant(nn.Module):
+    r"""The robust Tolerant kernel cost function.
+
+    .. math::
+        \bm{y}_i = b\log (1+e^{\frac{\bm{x}_i-a}{b}})-b\log (1+e^{\frac{-a}{b}})
+
+    where :\bm{a}  and \bm{b} are hyperparameters, :math:`\bm{x}` and :math:`\bm{y}` are the
```

Comment:
the math display has issues: 
<img width="235" alt="image" src="https://user-images.githubusercontent.com/8695500/191582867-9cca8347-06fc-4c23-8d9d-553598fc14b1.png">
you forget `:math`